eis_toolkit/prediction/weights_of_evidence.py

```diff
@@ -0,0 +1,255 @@
+from numbers import Number
+
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+import rasterio
+from beartype import beartype
+from beartype.typing import List, Literal, Optional, Sequence, Tuple
+
+from eis_toolkit import exceptions
+from eis_toolkit.vector_processing.rasterize_vector import rasterize_vector
+
+
+def _read_and_preprocess_evidence(raster: rasterio.io.DatasetReader, nodata: Optional[Number] = None) -> np.ndarray:
+    """Read raster data and handle NoData values."""
+
+    array = np.array(raster.read(1), dtype=np.float32)
+
+    if nodata is not None:
+        array[array == nodata] = np.nan
+    elif raster.meta["nodata"] is not None:
+        array[array == raster.meta["nodata"]] = np.nan
+
+    return array
+
+
+def _calculate_metrics_for_class(deposits: np.ndarray, evidence: np.ndarray) -> tuple:
+    """Calculate weights/metrics for given data."""
+    A = np.sum(np.logical_and(deposits == 1, evidence == 1))
+    B = np.sum(np.logical_and(deposits == 1, evidence == 0))
+    C = np.sum(np.logical_and(deposits == 0, evidence == 1))
+    D = np.sum(np.logical_and(deposits == 0, evidence == 0))
+
+    if A + B == 0:
+        raise Exception("No deposits")
```

Comment:
I suggest a custom `Exception` or at least a more specific exception like `ValueError`.

Reply:
I actually modified this so that `A == 0` and ` C + D == 0` lead to skipping calculating metrics for the class instead of raising an exception. This is more the behavior that is wanted if I'm not mistaken (I could since I'm don't know the model well).